cluster-provision/images/vm-image-builder/customize-image.sh

```diff
@@ -1,14 +1,27 @@
 #!/usr/bin/env bash
 set -exuo pipefail
 
+ARCH=${ARCH:-"x86_64"}
+
 function cleanup() {
   if [ $? -ne 0 ]; then
     rm -f "${CUSTOMIZE_IMAGE_PATH}"
+    virsh destroy "${DOMAIN_NAME}" || true
+    undefine_vm "${DOMAIN_NAME}" "${ARCH}"
   fi
 
   rm -rf "${CLOUD_INIT_ISO}"
-  virsh destroy "${DOMAIN_NAME}" || true
-  virsh undefine "${DOMAIN_NAME}" || true
```

Comment:
Note that there is another case, when the script is interrupted (e.g: CTRL+C) .
The trap will catch the SIGINT signal and trigger a clean up, but the VM wont be removed because the return code is 0 ($? = 0) - thus cleanup() should remove the VM regardless of the script return code.

Reply:
@zhlhahaha

Reply:
> Note that there is another case, when the script is interrupted (e.g: CTRL+C) . The trap will catch the SIGINT signal and trigger a clean up, but the VM wont be removed because the return code is 0 ($? = 0) - thus cleanup() should remove the VM regardless of the script return code.

Ok, I see, thanks.